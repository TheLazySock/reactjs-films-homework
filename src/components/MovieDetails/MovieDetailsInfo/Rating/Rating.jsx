import React from 'react';
import PropTypes from 'prop-types';
import style from './Rating.scss';

const Rating = (props) => {
  const { rating } = props;
  const filled = Array.from(
    Array(Math.round(rating)),
    (elem, i) => <li className={style.star} key={`filled-${i}`}><span>&#9733;</span></li>,
  );
  const empty = Array.from(
    Array(5 - filled.length),
    (elem, i) => <li className={style.star} key={`empty-${i}`}><span>&#9734;</span></li>,
  );

  return (
    <div className={style.rating}>
      <ul className={style.starsList}>
        {filled}
        {empty}
      </ul>
      <span className={style.ratingAverage}>{rating}</span>
    </div>
  );
};

Rating.propTypes = {
  rating: PropTypes.number.isRequired,
};

export default Rating;
