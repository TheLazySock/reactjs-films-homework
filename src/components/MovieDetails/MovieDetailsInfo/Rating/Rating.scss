.rating {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
}

.starsList {
  margin: 0;
  padding: 5px 0;
  list-style: none;
  display: flex;
  font-size: 24px;
  line-height: 22px;
  color: #0aaee4;
}

%star {
  padding-right: 5px;
}

.filledStar {
  @extend %star;

  &:after {
    content: '\2605';
  }
}

.emptyStar {
  @extend %star;

  &:after {
    content: '\2606';
  }
}

.ratingAverage {
  margin: auto;
  margin-left: 5px;
  padding: 0 7px;
  font-size: 13px;
  line-height: 18px;
  color: white;
  border: 1px solid #0aaee4;
  border-radius: 5px;
}
