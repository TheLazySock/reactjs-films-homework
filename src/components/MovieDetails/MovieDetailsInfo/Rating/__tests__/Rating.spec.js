import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Rating from '../index';

describe('Rating', () => {
  it('renders properly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Rating rating={3} />);
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
  });
});
