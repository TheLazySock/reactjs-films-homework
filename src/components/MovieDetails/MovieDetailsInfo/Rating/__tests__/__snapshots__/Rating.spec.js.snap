// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders properly 1`] = `
<div
  className="rating"
>
  <ul
    className="starsList"
  >
    <li
      className="filledStar"
    />
    <li
      className="filledStar"
    />
    <li
      className="filledStar"
    />
    <li
      className="emptyStar"
    />
    <li
      className="emptyStar"
    />
  </ul>
  <span
    className="ratingAverage"
  >
    3
  </span>
</div>
`;
